pkg/api/validation/validation.go

```diff
@@ -1252,26 +1252,56 @@ func validateEndpointSubsets(subsets []api.EndpointSubset) errs.ValidationErrorL
 
 	for i := range subsets {
 		ss := &subsets[i]
-
-		ssErrs := errs.ValidationErrorList{}
-
 		if len(ss.Addresses) == 0 {
-			ssErrs = append(ssErrs, errs.NewFieldRequired("addresses"))
+			allErrs = append(allErrs, errs.NewFieldRequired("addresses"))
 		}
 		if len(ss.Ports) == 0 {
-			ssErrs = append(ssErrs, errs.NewFieldRequired("ports"))
+			allErrs = append(allErrs, errs.NewFieldRequired("ports"))
+		}
+		for addr := range ss.Addresses {
+			allErrs = append(allErrs, validateEndpointAddress(&ss.Addresses[addr])...)
 		}
-		// TODO: validate each address and port
-		allErrs = append(allErrs, ssErrs.PrefixIndex(i)...)
```

Comment:
I think you've now lost the index - what does the error for this come out as?  It should come out as (for example) "subsets[0].ports[3].name"